src/routes/autorag.ts

```diff
@@ -0,0 +1,335 @@
+import type { Context } from "hono";
+import type { Env } from "../middleware/auth";
+import type { AuthPayload } from "../services/auth-service";
+import { AUTORAG_CONFIG } from "../shared";
+
+// AutoRAG API response types
+interface AutoRAGSyncResponse {
+  success: boolean;
+  result: {
+    job_id: string;
+    message?: string;
+  };
+}
+
+interface AutoRAGJobResponse {
+  success: boolean;
+  result: {
+    id: string;
+    source: string;
+    started_at: string;
+    last_seen_at?: string;
+    ended_at?: string;
+    end_reason?: string;
+  };
+}
+
+interface AutoRAGJobLogsResponse {
+  success: boolean;
+  result: {
+    logs: Array<{
+      timestamp: string;
+      level: string;
+      message: string;
+    }>;
+  };
+}
+
+interface AutoRAGJobsResponse {
+  success: boolean;
+  result: {
+    jobs: Array<{
+      id: string;
+      source: string;
+      started_at: string;
+      last_seen_at?: string;
+      ended_at?: string;
+      end_reason?: string;
+    }>;
+  };
+}
+
+// Extend the context to include userAuth
+type ContextWithAuth = Context<{ Bindings: Env }> & {
+  userAuth?: AuthPayload;
+};
+
+/**
+ * Trigger AutoRAG sync for a specific RAG resource
+ * This endpoint triggers the Cloudflare AutoRAG service to process a file
+ */
+export async function handleAutoRAGSync(c: ContextWithAuth) {
+  try {
+    const ragId = c.req.param("ragId");
+    const username = (c as any).userAuth.username;
+
+    console.log(
+      `[AutoRAG] Triggering sync for RAG ID: ${ragId}, user: ${username}`
+    );
+
+    // Call the Cloudflare AutoRAG sync API
+    const accountId = c.env.AUTORAG_ACCOUNT_ID;
+    const apiUrl = c.env.AUTORAG_API_URL;
+
+    if (!accountId || !apiUrl) {
+      throw new Error("AutoRAG configuration missing: ACCOUNT_ID or API_URL");
+    }
+
+    // Use a consistent RAG ID for the library instead of the filename
+    const syncUrl = AUTORAG_CONFIG.buildLibraryAutoRAGUrl(
+      accountId,
+      apiUrl,
+      "/sync"
+    );
+    console.log(`[AutoRAG] Calling sync API: ${syncUrl}`);
+
+    const response = await fetch(syncUrl, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${c.env.AUTORAG_API_TOKEN}`,
+      },
+    });
+
+    if (!response.ok) {
+      const errorText = await response.text();
+      throw new Error(
+        `AutoRAG sync API failed: ${response.status} ${errorText}`
+      );
+    }
+
+    const result = (await response.json()) as AutoRAGSyncResponse;
+    console.log(`[AutoRAG] Sync API response:`, result);
+
+    if (!result.success || !result.result?.job_id) {
+      throw new Error("Invalid response from AutoRAG sync API: missing job_id");
```

Comment:
technically this error may or may not be correct since we're checking two conditions